packages/block-library/src/post-excerpt/index.php

```diff
@@ -84,8 +84,7 @@ function register_block_core_post_excerpt() {
  * Returns 100 because 100 is the max length in the setting.
  */
 if ( is_admin() ||
-	defined( 'REST_REQUEST' ) ||
-	'REST_REQUEST' ) {
+	defined( 'REST_REQUEST' ) && REST_REQUEST ) {
```

Comment:
Even if this condition is no longer causing those false positives for REST_REQUEST, this is still a very concerning piece of code. This effectively _always_ overrides `excerpt_length` regardless of its source. See [my comment here](https://github.com/WordPress/gutenberg/pull/44964#discussion_r1122855365).